examples/grid2grid/main.py

```diff
@@ -0,0 +1,161 @@
+
+            with ChargingBar('Computing ' + config['grids']['names_verbose'][grid_iter] + ' grid', max=len(_grid_points)) as bar:
+                for _counter in range(len(_grid_points)):
+                    x = _grid_points[_counter][0]/1000
+                    y = _grid_points[_counter][1]/1000
+                    z = (_grid_points[_counter][2]+ee_h+obj_h)/1000  # [m]
+                    pos_obj = [x, y, z]  # [m]
+                    pos_obj_hover = [x, y, z + hover_h]  # [m]
+                    extra_angle = config['grids']['angle_pickup'][grid_iter]     # Additional pickup angle
+                    success_pick, _joints_pick = solve_ik(self.eva, guess, extra_angle, pos_obj)
+                    success_hover, _joints_hover = solve_ik(self.eva, guess, extra_angle, pos_obj_hover)
+                    if ('success' not in success_pick) or ('success' not in success_hover):
+                        raise Exception('IK error')  # Failed IK. Position not reachable
+                    _joints[grid_iter]['pick'].append(_joints_pick)
+                    _joints[grid_iter]['hover'].append(_joints_hover)
+                    bar.next()
+                if self.plot:
+                    plot_grids(_grid_x, _grid_y, _grid_z, ax, grid_iter)
+        if self.plot:
+            plt.show()
+        move_eva = input("Please verify the correctness of grid placement before continuing "
+                         "(to do this, set plot_on_off variable to True and verify grids placement from the plots).\n"
+                         "Proceed (this will move the robot)? yes/no\n")
+        if 'yes' not in move_eva:
+            raise Exception('\n\n\nScript aborted by user: \n - set the plot_on_off variable to True\n '
+                            '- run the script again\n - verify grids placement from the plots ')
+        return _joints
+
+
+if __name__ == "__main__":
+    # Load use-case parameters
+    config = load_use_case_config()
+
+    # Connection to robot
+    host = config['EVA']['comm']['host']
+    token = config['EVA']['comm']['token']
+    eva = Eva(host, token)
+
+    # Compute grid points and robot joints
+    plot_on_off = True
+    eva_box = EvaGrids(eva, plot_on_off)
+    joints = eva_box.get_grid_points(config['grids']['names'])
+
+    # Go home before starting
+    with eva.lock():
+        eva.control_go_to(config['joints']['home'])
+
+    while True:
+        for counter in range(len(joints[(config['grids']['names'][0])]['pick'])):
+            joints_home = config['EVA']['home']
+            joints_pick = joints[config['grids']['names'][0]]['pick'][counter]
+            joints_drop = joints[config['grids']['names'][1]]['pick'][counter]
+            joints_pick_hover = joints[config['grids']['names'][0]]['hover'][counter]
+            joints_drop_hover = joints[config['grids']['names'][1]]['hover'][counter]
+
+            # USER DEFINED WAY-POINTS
+            joints_operation_A = []

```

Comment:
Not sure what these three are for?

Reply:
these are user defined waypoints, to be added between the origin and target frames (i.e. any operation to be done with the objects when they are gripped)